stacker/exceptions.py

```diff
@@ -17,6 +17,16 @@ def __init__(self, lookup, *args, **kwargs):
         super(UnknownLookupType, self).__init__(message, *args, **kwargs)
 
 
+class InvalidUserdataPlaceholder(Exception):
+
+    def __init__(self, blueprint_name, *args, **kwargs):
+        message = "Could not parse userdata in blueprint \"%s\". " % (
```

Comment:
I'd include the text from the ValueError that raises this - would make it easier to track down the issues:

```
>>> t.substitute({"100": "blue", "a": "apple"})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python/2.7.12_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/string.py", line 176, in substitute
    return self.pattern.sub(convert, self.template)
  File "/usr/local/Cellar/python/2.7.12_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/string.py", line 173, in convert
    self._invalid(mo)
  File "/usr/local/Cellar/python/2.7.12_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/string.py", line 146, in _invalid
    (lineno, colno))
ValueError: Invalid placeholder in string: line 2, col 1
```

The line # & col # seem useful.  You can get that from `.args` of the original error.